src/Bowling.js

```diff
@@ -0,0 +1,158 @@
+console.log("testtest");
+function BowlingGame() {
+  this.framesLeft = 10;
+  this.STRIKE = 10;
+
+  this.currentRoll = 0;
+  this.currentScore = 0;
+  this.totalScore = null;
+
+  this.isStrike = false;
+  this.strikeCount = 0;
+  this.strikeScore = 0;
+  this.consecutiveStrikes = 0;
+
+  this.isSpare = false;
+  this.spareCount = 0;
+  this.spareScore = 0;
+
+  this.lastFrameExtra = false;
+
+  this.GAMEOVER_ERROR = "The game is over you cannot bowl again";
+
+  };
+
+  BowlingGame.prototype.bowl = function(pins) {
+    if ( this.framesLeft === 0 ) throw Error(this.GAMEOVER_ERROR);
+
+    if (pins === this.STRIKE) { this.isStrike = true;}
+    if (((this.currentScore + pins) === 10) && this.currentRoll === 1) { this.isSpare = true;}
+
+    if ( this.framesLeft === 1 ) { this.lastFrameCheck(pins);
+    }
+
+    if (this.lastFrameExtra === true) {
+      this.lastFrame(pins);
+      } else {
+      this.currentMove(pins);
+      }
+  };
+
+
+  BowlingGame.prototype.currentMove = function(pins) {
```

Comment:
This function has too many statements. (11)<br>This function's cyclomatic complexity is too high. (6)